### PR TITLE
Pull requests should target master branch

### PR DESCRIPTION
**Pull request by JoelMarcey**
Make it more obvious that pull requests should generally target the `master` branch. Provide visible clarity in both the `CONTRIBUTING.md` and the actual pull request template.

Test Plan:

- markdown renders correctly
- eyes

### CODE REVIEW

**Comment by facebook-github-bot-0** [comment]
By analyzing the blame information on this pull request, we identified @zxcpoiu and @mkonicek to be potential reviewers.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 Thanks for submitting a pull request! Please provide enough information so that others can review your pull request:
 
+> **Unless you are cherry-picking an existing commit into a current release, ensure your pull request is targeting the `master` React Native branch.**
```
Comment (ide) [inline .github/PULL_REQUEST_TEMPLATE.md]: maybe add "Unless you are [a React Native release maintainer and] cherry-picking..." or some other wording so it's clear that nearly all contributors never need to cherry pick.

**Comment by JoelMarcey** [comment]
@ide Are we good here now?

**Comment by ide** [comment]
Thanks!

@facebook-github-bot shipit

**Comment by facebook-github-bot-4** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1086196974804884/int_phab) to review internal test results.

**Pull request closed without merging**